test/sequential/sequencerStaking.cancel.test.ts

```diff
@@ -0,0 +1,403 @@
+    n--;
+  }
+}
+
+async function createACollatorUser() {
+  const [user] = setupUsers();
+  await Sudo.asSudoFinalized(Assets.mintNative(user));
+  return user;
+}
+
+async function leaveSequencingIfAlreadySequencer(userAddr: string) {
+  const stakedEth = await SequencerStaking.sequencerStake(userAddr, "Ethereum");
+  const stakedArb = await SequencerStaking.sequencerStake(userAddr, "Arbitrum");
+  let chain = "";
+  if (stakedEth.toHuman() !== "0") {
+    chain = "Ethereum";
+  } else if (stakedArb.toHuman() !== "0") {
+    chain = "Arbitrum";
+  }
+  if (chain !== "") {
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+    await Sudo.asSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        userAddr,
+        await SequencerStaking.leaveSequencerStaking(chain as ChainName),
+      ),
+    );
+    await Sudo.asSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        userAddr,
+        await SequencerStaking.unstake(chain as ChainName),
+      ),
+    );
+  }
+}
+
+async function createAnUpdateAndCancelIt(
+  seq: User,
+  canceler: User,
+  chain: ChainName = "Ethereum",
+  cancellation: boolean = true,
+) {
+  const seqAddress = seq.keyRingPair.address;
+  const cancelerAddress = canceler.keyRingPair.address;
+  await Rolldown.waitForReadRights(seqAddress, 50, chain);
+  const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+  const api = getApi();
+  const update = new L2Update(api)
+    .withDeposit(txIndex, seqAddress, seqAddress, BN_MILLION)
+    .on(chain)
+    .build();
+  let reqId = 0;
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(seqAddress, update),
+  ).then(async (events) => {
+    await waitSudoOperationSuccess(events, "SudoAsDone");
+    reqId = Rolldown.getRequestIdFromEvents(events);
+  });
+  //const disputePeriodStartBlock = await getBlockNumber();
+  let reqIdCanceled: number = 0;
+  if (cancellation === true) {
+    const cancel = await Sudo.asSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        cancelerAddress,
+        await Rolldown.cancelRequestFromL1(chain, reqId),
+      ),
+    );
+    await waitSudoOperationSuccess(cancel, "SudoAsDone");
+    reqIdCanceled = Rolldown.getRequestIdFromCancelEvent(cancel);
+  }
+  return { txIndex, api, reqId, reqIdCanceled };
+}
+
+const preSetupSequencers = {
+  Ethereum: "0x3cd0a705a2dc65e5b1e1205896baa2be8a07c6e0",
+  Arbitrum: "0x798d4ba9baf0064ec19eb4f0a1a45785ae9d6dfc",
+};
+
+beforeAll(async () => {
+  await initApi();
+  setupUsers();
+  await setupApi();
+  //Add a few tokes because some tests may end up on slashing them
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      preSetupSequencers.Ethereum,
+      await SequencerStaking.provideSequencerStaking(BN_ZERO, "Ethereum"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      preSetupSequencers.Arbitrum,
+      await SequencerStaking.provideSequencerStaking(BN_ZERO, "Arbitrum"),
+    ),
+  );
+});
+
+beforeEach(async () => {
```

Comment:
same, plz if you duplicate some code, refactor it first.

Reply:
Refactored